content/Rails N + 2 queries - Bruno Vezoli.md

```diff
@@ -0,0 +1,223 @@
+## Rails N + 2 queries
+
+We usually try to solve performance problems by using `#includes` to get rid of N + 1 queries but this doesn't
+always get rid of them; in fact it can even create more queries under some circumstances. This post
+shows some examples of when that happens and how to deal with it.
+
+![Forest with bridge](images/forest_bridge.jpg)
+
+### Your typical N + 1 query problem
+
+We learn how to deal with ActiveRecord performance issues with very plain examples, most blog posts show models that
+look something like this:
+
+```ruby
+class Post < ApplicationRecord
+  has_many :comments
+end
+
+class Comment < ApplicationRecord
+  belongs_to :post
+end
+```
+
+Followed by your standard `irb` example:
+
+```ruby
+irb> posts = Post.all
+irb> all_comments = posts.map { |p| p.comments }
+```
+
+Which generates the following queries, where we can clearly see an N + 1 query since it needs to fetch comments for each
+post:
```

Comment:
As this is the gist of N+1, I'll complete this sentence with something emphasizing that.

```suggestion
Which generates the following queries where we can clearly see the infamous N+1 problem: one query for retrieving the posts and N additional queries for retrieving the comments, where N is the number of posts.
```